### PR TITLE
Create initrd-root-device.target synchronization point

### PR DESCRIPTION
**Pull request by dsd**
Add a synchronization point so that custom initramfs units can run
after the root device becomes available, before it is fsck'd and
mounted.

This is useful for custom initramfs units that may modify the
root disk partition table, where the root device is not known in
advance (it's dynamically selected by the generators).

### CODE REVIEW

File: src/fstab-generator/fstab-generator.c [hunk 1]
```diff
@@ -508,6 +509,13 @@ static int add_sysroot_mount(void) {
                 opts = arg_root_options;
 
         log_debug("Found entry what=%s where=/sysroot type=%s", what, strna(arg_root_fstype));
+
+        if (is_device_path(what)) {
+            r = generator_write_initrd_root_device_deps(arg_dest, what);
+            if (r < 0)
+                    return r;
```
Comment (poettering) [inline src/fstab-generator/fstab-generator.c]: 8ch indenting plesae, see CODING_STYLE.

File: src/gpt-auto-generator/gpt-auto-generator.c [hunk 2]
```diff
@@ -956,6 +956,12 @@ static int add_root_mount(void) {
          * wait for a root device to show up. A udev rule will create
          * the link for us under the right name. */
 
+        if (in_initrd()) {
+            r = generator_write_initrd_root_device_deps(arg_dest, "/dev/gpt-auto-root");
+            if (r < 0)
+                    return 0;
+        }
+
```
Comment (poettering) [inline src/gpt-auto-generator/gpt-auto-generator.c:964]: indenting... see CODING_STYLE.

File: src/shared/generator.c [hunk 3]
```diff
@@ -191,3 +191,18 @@ int generator_write_timeouts(
                                     "[Unit]\nJobTimeoutSec=%s",
                                     program_invocation_short_name, timeout);
 }
+
+int generator_write_initrd_root_device_deps(const char *dir, const char *what)
+{
+        _cleanup_free_ char *unit = NULL;
```
Comment (poettering) [inline src/shared/generator.c:196]: please follow coding style. opening bracket should be on same line as function name...

**Comment by poettering** [comment]
after conversing with @haraldh this looks Ok otherwise.

**Comment by dsd** [comment]
Thanks for the quick feedback. I pushed an updated commit, the only changes are those 3 coding style fixes.

**Pull request merged**